### ISSUE TITLE
Cannot see Citizen UI - Citizens in default skin / 

### ISSUE THREAD

**Issue by asy-m** [ldtteam/minecolonies#8411]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Are you using the latest MineColonies Version?

- [X] I am running the latest **alpha** version of MineColonies for my Minecraft version.
I am also running the latest (possible) versions of other mods that are part of my problem.

### Did you check on the Wiki? or ask on Discord?

- [X] I checked the MineColonies/Structurize [Wiki](https://wiki.minecolonies.ldtteam.com/) or I asked on [discord](https://discord.minecolonies.com/). Or I was sent from discord to open an issue here.

### Minecraft Version

1.16

### MineColonies Version

1.16.5-1.0.770-Beta

### Structurize Version

Structurize-1.16.4-1.0.392-Release

### Related Mods and their Versions

Forge-1.16.5-36.2.31

Modpack: Project Architect 1.9
https://www.curseforge.com/minecraft/modpacks/project-architect/files/3693929

### Current Behavior

I cannot interact with the citizens. The citizens are using the default skin, and default name colors.  The citizens are using different names, however.

The other player on the server is able to interact with the citizen, sometimes. If the player leaves the area, and returns, it is likely they will be unable to interact with the citizen as well. I have been unable to interact with the citizen UIs at all. It seems the other players are luckier at getting to interact than I am. 

![image](https://user-images.githubusercontent.com/106496182/170894927-8bdf6315-c635-4037-9edb-67bbfe8f88a2.png)

### Expected Behavior

I expect to be able to right click a citizen, and interact with their UI for needs / inventory etc.
I expect to be able to see the chosen nametag color.
I expect to see the citizens using different models, instead of what I assume to be the default skin.

### Reproduction Steps

The colony we set up is the Warped Colony in the Overworld.

I think it has to do with leaving and unloading the MC chunks, and when I return, I am unable to access their UI. Occasionally, it can work whenever I return, although it typically only works on one or two citizens.

The other player has had the same issue, but for whatever reason seems to be luckier in getting the UI to work after leaving the area / returning.  I have no such luck and the other player has for now, just stayed within the town and not left nearby chunks.

### Logs

-

### Anything else?

I added MineColonies to the modpack for both the server and clients.  We originally used the Release version, but updated to 1.16.5-1.0.770-Beta as a step in attempting to fix. Still using Structurize-1.16.4-1.0.392-Release. 

We are not using Optifine, or Oculus in the modpack. We are using Graphics: Fancy. 

### Footer

---
## Viewers
* Add a thumbs-up to the bug report if you are also affected. This helps the bug report become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.

**Comment by Raycoms** [comment]
How long since you restarted?

**Comment by asy-m** [comment]
I had restarted a few hours ago in an attempt to fix things.  But I went ahead and completed a restart on both the client and the server. 

When I load in - a few of the citizens are interactable, but not all.  The typical loop that happens is over time the citizens slowly become bugged as I play.

As I typed this, I jumped into the Nether, then back, and most appear bugged again.

**Comment by Raycoms** [comment]
can you send us your log?

**Comment by asy-m** [comment]
Sorry - I don't know what kind of information is contained in the log, so I'm not sure I'm comfortable posting it.

**Comment by Raycoms** [comment]
Nothing to worry about, those logs are made to not contain any "sensible" info. At most it shows you the folder path where it's located. you can also PM me the log on discord if you feel more comfy that way.

**Comment by Brad2003HUD0** [comment]
Hey I'm having the exact same issue, did you ever resolve this?

**Comment by asy-m** [comment]
Exact same issue with MineColonies and Project Architect?

TLDR:
I ended up having to run this specific setup as a LAN world instead, and having other players connect to me that way.

Also:
 We had another issue, although I'm not sure if it was specifically MineColonies or something in Project Architect where the other player would disconnect if they logged out in the town and near the base. So we instead would have them log out somewhere away from the base, and when they decided to join the game, I would go to that same location and we would walk into town together. I don't know why, but it seemed to help.

**Comment by asy-m** [comment]
Just for more details, I'll include some comments of what works we did to try to fix it. Ultimately, however, I was unable to get this issue resolved.

I was running Project Architect modpack, and I manually added MineColonies and Structurize to the server and the clients. My specific setup was I had two PCs running clients, and a third PC running the server.  

I was having this issue on brand new worlds, even when being the only person logged into the server. Players were having issues talking to the colonists independent of each other, so for instance I would not be able to talk to the colonist while the other players could (and vice versa). 

I talked to a developer for a bit, and they had theorized it was a client not getting correct data from the server, but the server seemed to be doing okay. The developer also tried recreating the issue, and was unsuccessful.  The developer however did not set up a PA+MC server and PA + MC client so I think they would not be able to see the issue. They tried a local setup and that worked, which is why I went down the LAN path myself.

The LAN game worked fine, but occasionally the other players would get some error that may or may not be related to MineColonies. It was a more generic error, and I don't have what the error is anymore.  I actually suspected it may have to do with Forge but I cannot really tell.  Anyway, the error said something loosely about transferring too much information, or an overflow error, something like that. The error would disconnect the player from the game.

What we found was that the other player had to log out somewhere where the main base, main colony was not chunk loaded. If they disconnected during the session, which rarely happened, I would quickly have to /tp the player to me, away from the main base chunks, and it would allow them to load again.  After the player would be stable, we would walk into town and it would generally be okay. I don't know what was happening, or why it worked, but there you go, in case you go down the same path.

Other than that, the world and game worked fine. In the LAN game, we ran a large MineColonies colony, and a decent base. I believe I even felt comfortable adding a couple other mods.

**Comment by Raycoms** [comment]
Were you able to reproduce this outside of the modpack?

Sent from Proton Mail mobile

-------- Original Message --------

**Comment by Brad2003HUD0** [comment]
So with us, I set a server up with Apex hosing and selected Engineers life 2, then I added minecolonies and structurize in manually into both the server and my client and my friend added it to his client pack too. Was working fine until we walked away from the colony and it was unloaded. When we try to approach the colonists they’re all the default skin and we can’t open their GUI, just get promoted with one of their voice lines. Seems it’s a client side issue, the colonists may be broke for me and not my friend… I found a fix which is to just pickup the colonist in creative mode with the Carry on mod, placing them back down resets them and fixes them. Not very practical as obviously don’t want to be playing creative all the time

**Comment by Sluethen** [comment]
Just adding on that I am also having this issue - It only appears on the server, creative single player is fine
It may be helpful to know that with the oneprobe mod, I cannot see the info for non-interactable npcs while I can for interactable NPC's

I have a server running that can consistantly reproduce this issue, DM me on discord @ Sluethen#0653 and I can get you setup with that (and logs)

**Comment by Raycoms** [comment]
Are you running any performance mods on the server?

-------- Original Message --------

**Comment by Sluethen** [comment]
I am running a mod called AI Improvements (not that this should be effecting it)
other then that, I don't see any mods that are performance related, I'm running the latest version of Project Architect with minecolonies added

**Comment by Sluethen** [comment]
and im running on the 1.16.5 1.0.878-BETA version

**Comment by Sluethen** [comment]
![2022-07-11_12 06 43](https://user-images.githubusercontent.com/10472251/178329454-f5742fc4-a4e7-4f74-96e7-d5cb257ecb01.png)
and this beautiful missing texture lumberjack (including this just in case its related)
